benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp/time_step_ompopt.cpp

```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+#include <chrono>
+#include <vector>
+#include <cstdlib>
+
+// Kernels
+#include "../../kernels/c_family/continuity_kern.c"
+#include "../../kernels/c_family/momentum_u_kern.c"
+#include "../../kernels/c_family/momentum_v_kern.c"
+#include "../../kernels/c_family/boundary_conditions_kern.c"
+#include "../../kernels/c_family/time_update_kern.c"
+
+// Optimizations:
+// - Fuse some kernels (x1.1), fusing more kernels didn't make any difference.
+// - Change array copies with swaps is x1.1 faster, but since the fortran->c
```

Comment:
I'm a bit confused here - the ptr-swapping code is commented out below and therefore this optimisation is not applied. If that is because of the limitations of fortran->c calling then please say so (here and below).

Reply:
That's right. I commented it instead of deleting it because I would still try to make it work in the future. As suggested I added comments here and below.